### DISCUSSION TITLE
Integrate an AuxVariable along a plane?

### DISCUSSION THREAD

**Discussion by mangerij** [idaholab/moose#16481] [Q&A General] [answered]
Hi,

I looked around and found ElementsAlongPlane and it seems you can get information about elements given some plane coordinate and plane normal. Is there a way to integrate an AuxVariable given the normal and coordinate? Also can this be block restricted?
thanks

**Comment by aeslaughter** [comment]
I don't think that capability exists, but can always be added.

**Comment by GiudGiud** [comment, marked as answer]
Hi,
You could add the Plane as a sideset in your mesh then use one of the integral postprocessors.

**Comment by mangerij** [comment]
We have tried that. It is a bit tricky when using tets and your plane cuts at weird angles in the mesh. And then you end up getting extra nodes near that desired sideset which give different solutions.

On Dec 22, 2020 1:10 AM, Guillaume Giudicelli <notifications@github.com> wrote:

*Message sent from a system outside of UConn.*


Hi,
You could add the Plane as a sideset in your mesh then use one of the integral postprocessors.

**Reply by GiudGiud** [reply]
Ok if you dont want it to influence your solution but are open to adding the plane in a mesh, then I suggest you use a Multiapp and transfer the solution fields to this other app. You can still use the original mesh in the initial app.
You can use the Transfers to interpolate the solution on the new mesh, and you can use the postprocessor on that mesh to obtain the desired integral.
